### PR TITLE
61120 Add Exception class for CHIP lib

### PR DESCRIPTION
**Pull request by randomsync**
## Summary
This PR adds an exception class for CHIP lib, which can be used when raising exceptions due to downstream errors.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#61120

## Testing done

- rspecs

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: config/locales/exceptions.en.yml [hunk 1]
```diff
@@ -2339,3 +2339,12 @@ en:
...
         status: 422
+
+  chip:
+    exceptions:
+      unmapped_service_exception:
+        <<: *defaults
+        title: 'Operation failed'
+        detail: The upstream server returned an error code that is unmapped
+        code: unmapped_service_exception
+        status: 400
```
Comment (kanchanasuriya) [inline config/locales/exceptions.en.yml:2350]: Is it possible to have this defined in a separate `exceptions.en.yml` file specific to chip lib?

Comment (kanchanasuriya) [reply config/locales/exceptions.en.yml:2350]: Discussed offline; lib seems to be referring to `config/locales/exceptions.en.yml`

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Review by ryan-mcneil — APPROVED** [review]
👌

**Pull request merged**